ui/lib/components/TreeDiagram/TreeDiagram.tsx

```diff
@@ -0,0 +1,532 @@
+import React, { useEffect, useState, useRef } from 'react'
+import { TreeProps } from './types'
+import { RawNodeDatum, TreeNodeDatum, Translate } from './types'
+import { hierarchy, HierarchyPointNode, HierarchyPointLink } from 'd3-hierarchy'
+
+import { flextree } from 'd3-flextree'
+
+import * as d3 from 'd3'
+import _ from 'lodash'
+import { select, event } from 'd3-selection'
+import { zoom as d3zoom, zoomIdentity } from 'd3-zoom'
+import { v4 as uuidv4 } from 'uuid'
+
+import LinksWrapper from './LinksWrapper'
+import NodesWrapper from './NodesWrapper'
+
+import styles from './index.module.less'
+import { Button } from 'antd'
+
+const TreeDiagram = (props: TreeProps) => {
```

Comment:
How about splitting TreeDiagram and TreeDiagramView. So that we can earn a pure TreeDiagram without custom Link, Node, and the third-part UI lib like `antd`.